### ISSUE TITLE
iOS orientation not being listened to

### ISSUE THREAD

**Issue by jpike88** [software-mansion/react-native-screens#1540]
@kkafar You guys are missing `[[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications]` from this library's initialisation, meaning that reported orientations on my test device (iOS 15, iPhone 11 Pro) kept returning `UIDeviceOrientationUnknown` from `[UIDevice currentDevice].orientation`, causing `enforceDesiredDeviceOrientation` to function incorrectly. The result was my device didn't change orientation when rotated, even though Xcode settings let any orientation happen.

Added the above code to my AppDelegate fixed this issue.

**Comment by jpike88** [comment]
I don't have the time to go beyond my initial post. That should contain enough information to act

**Comment by kkafar** [comment]
Hi @jpike88,
I've tried to reproduce this on fresh application with both simulator & physical device (iOS 15.5) and failed. The values returned from call to `[[UIDevice currentDevice] orientation]` seem to be correct, and *navigating between screens with different orientations* seems to work correctly. I must admit however, that neither we call `[[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications]` nor any dependency does this for us (I looked up in `node_modules` with `grep --include=\*.{m,mm} -rnw './node_modules/' -e "beginGeneratingDeviceOrientationNotifications"`), so [according to the docs](https://developer.apple.com/documentation/uikit/uidevice/1620053-orientation?language=objc) the code should not work, as you say. Looking from my POV: docs are lying or there is different interaction causing our code to work by accident. I'll issue PR with requested changes, but will wait with merging it until we understand whats going on.
